File: common/src/main/java/com/common/listeners/NewMapListener.java
package com.common.listeners;

import com.common.Data;
import com.common.graphics.TextInput;
import com.common.menu.MenuCommands;

import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.io.IOException;

import static com.common.Data.fileHandle;

public class NewMapListener implements ActionListener {
	/**
	 * Text input listener for text input for creating new game
	 *
	 * @param e the event to be processed
	 */
	@Override
	public void actionPerformed(ActionEvent e) {
		try {
			String map_name = TextInput.getText();
			String new_map = fileHandle.loadText("json/templates/map.json", true);
			fileHandle.saveText("/maps/" + map_name + ".json", new_map);
			Data.Map.currentMap = "map";
			TextInput.dispose();
			MenuCommands.newMapLevelEdit(map_name);
		} catch (IOException ex) {
			throw new RuntimeException(ex);
		}
	}
}

File: composeApp/src/commonMain/kotlin/com/one_of_many_simons/one_button_game/listeners/NewMapListener.kt
package com.one_of_many_simons.one_button_game.listeners

import com.one_of_many_simons.one_button_game.Data
import com.one_of_many_simons.one_button_game.Debug.Flags.Listeners.NEW_MAP_LISTENER
import com.one_of_many_simons.one_button_game.Debug.Levels.CORE
import com.one_of_many_simons.one_button_game.Debug.debug
import com.one_of_many_simons.one_button_game.Libraries.fileHandle
import com.one_of_many_simons.one_button_game.Libraries.menuCommands
import com.one_of_many_simons.one_button_game.Libraries.textInputListeners
import java.io.IOException

/**
 * Listener for new map name input from text input listener
 */
fun newMapListener() {
    debug(NEW_MAP_LISTENER, CORE, "--- [newMapListener]")

    try {
        val mapName = textInputListeners.getText().replace("\n", "")
        val newMap: String = fileHandle.loadText("json/templates/map.json", true)!!
        fileHandle.saveText("/maps/$mapName.json", newMap)
        Data.Map.currentMap = mapName
        menuCommands.newMapLevelEdit(mapName)

        textInputListeners.hide()
    } catch (ex: IOException) {
        throw RuntimeException(ex)
    }
}